content/Encapsulation/Encapsulation.q.md

```diff
@@ -0,0 +1,1470 @@
+# Encapsulation
+
++++
+
+### Part 1: Sample Code Analysis
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+class Person
+{
+    private $name = "Anna";
+    private $age;
+
+    public function setName($newName)
+    {
+        $this->name = $newName;
+    }
+
+    public function setAge($newAge)
+    {
+        if ($this->checkAge($newAge)) {
+            $this->age = $newAge;
+        }
+    }
+
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    public function getAge()
+    {
+        return $this->age;
+    }
+
+    private function checkAge($age)
+    {
+        if ($age > 0) {
+            return true;
+        }
+        return false;
+    }
+}
+
+$personObject = new Person();
+$personObject->setAge(12);
+echo $personObject->getName() . " is " . $personObject->getAge() . " years old.";
+?>
+```
+/// type=SS, id=a59f5f40-d624-40ed-b24a-d14c916cf189, answer=[5]
+
+Execute the program. What is its output?
+
+ - It prints `12`.
+
+ - It prints `Anna`.
+
+ - It produces an error.
+
+ - No output is displayed.
+
+ - It prints `Anna is 12 years old.`.
+
+
+/// type=MS, id=f7ae3a83-08e4-496e-bb5d-0ee32c7028f5, answer=[1,2]
+
+Which of the following are properties?
+
+ - `$age`
+
+ - `$name`
+
+ - `getAge()`
+
+ - `getName()`
+
+ - `$personObject`
+
+
+/// type=MS, id=4895378f-27f3-4aa3-987f-6da746a8862e, answer=[2,3,4]
+
+Which of the following are methods?
+
+ - `$name`
+
+ - `getAge()`
+
+ - `setName()`
+
+ - `checkAge()`
+
+ - `$personObject`
+
+
+/// type=SS, id=5d2004da-af63-4578-8847-076ff7901b3d, answer=[4]
+
+Which of the following is a private method?
+
+ - `$name`
+
+ - `getAge()`
+
+ - `setName()`
+
+ - `checkAge()`
+
+ - `$personObject`
+
+
+/// type=MS, id=c40bd1a5-10d4-4ff3-ada2-27f4847a0433, answer=[3,4]
+
+Which of the following are visibility keywords?
+
+ - `new`
+
+ - `class`
+
+ - `public`
+
+ - `private`
+
+ - `function`
+
+
+/// type=MS, id=5f5631b3-5797-417c-98f4-22f9bc55d232, answer=[1,4]
+
+In the statement `private $name = "Anna";` on line 4, what does `private` do?
+
+ - It encapsulates the `$name` property of the `Person` class.
+
```

Comment:
`private $name = "Anna"` is not encapsulation yet. But once it is enclosed in a method, that's the time it becomes Encapsulation

Reply:
This is one of my basis for having this statement.
![image](https://user-images.githubusercontent.com/25945790/77599933-61a54580-6f41-11ea-8b55-b9f9a3b9f315.png)

Reply:
Yes, the `private` restricts access to the property which is part of encapsulation.